homeassistant/components/mqtt/device_trigger.py

```diff
@@ -194,27 +209,19 @@ class AttachedTrigger:
 
     if DEVICE_TRIGGERS not in hass.data:
         hass.data[DEVICE_TRIGGERS] = {}
-    hass.data[DEVICE_TRIGGERS][discovery_id] = Trigger(
-        hass=hass,
-        device_id=device.id,
-        type=config[CONF_TYPE],
-        subtype=config[CONF_SUBTYPE],
-        topic=config[CONF_TOPIC],
-        payload=config[CONF_PAYLOAD],
-        qos=config[CONF_QOS],
-        discovery_id=discovery_id,
-    )
-
-    if ATTACHED_DEVICE_TRIGGERS not in hass.data:
-        return
-
-    # If someone was waiting for this trigger to be discovered, subscribe now
-    for trig in hass.data[ATTACHED_DEVICE_TRIGGERS]:
-        if trig.discovery_id == discovery_id:
-            device_trigger = hass.data[DEVICE_TRIGGERS][discovery_id]
-            trig.remove = await device_trigger.attach_trigger(
-                trig.action, trig.automation_info
-            )
+    if discovery_id not in hass.data[DEVICE_TRIGGERS]:
+        hass.data[DEVICE_TRIGGERS][discovery_id] = Trigger(
+            attached_triggers=[],
```

Comment:
Is this ever not an empty list? In that case just use a factory for the definition that creates a new list.